src/main/java/wooteco/subway/dao/line/JdbcLineDao.java

```diff
@@ -0,0 +1,83 @@
+package wooteco.subway.dao.line;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+import wooteco.subway.domain.Line;
+
+import java.sql.PreparedStatement;
+import java.util.List;
+import java.util.Optional;
+
+@RequiredArgsConstructor
+@Repository
+public class JdbcLineDao implements LineDao {
+
+    private final JdbcTemplate jdbcTemplate;
+
+    @Override
+    public Line save(Line line) {
+        String sql = "INSERT INTO line (name, color) VALUES (?, ?)";
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+        jdbcTemplate.update(con -> {
+            PreparedStatement preparedStatement = con.prepareStatement(sql, new String[]{"id"});
+            preparedStatement.setString(1, line.getName());
+            preparedStatement.setString(2, line.getColor());
+            return preparedStatement;
+        }, keyHolder);
+        return Line.of(keyHolder.getKey().longValue(), line.getName(), line.getColor(), line.getSections());
+    }
+
+    @Override
+    public Optional<Line> findByNameOrColor(String name, String color) {
+        String sql = "SELECT * FROM line WHERE (name = ? OR color = ?)";
+        try {
+            return Optional.ofNullable((Line) jdbcTemplate.queryForObject(sql, getRowMapper(), name, color));
```

Comment:
```suggestion
            return Optional.ofNullable(jdbcTemplate.queryForObject(sql, getRowMapper(), name, color));
```

`queryForObject()`는 형변환이 불필요한 메서드네요~